Reorganize man pages to create chapter 7 pages for the GEOPM PIO interface

cmcantalupo opened the pull request:
- Includes Asma's work to add an SST man page for chapter 7
- Fixes #2314

Commit:
Added geopm_pio_sst.7 man page

- Describes the signals and controls provided by the
  SSTIOGroup.

Signed-off-by: Asma H Al-Rawi<asma.h.al-rawi@intel.com>

cmcantalupo committed:
Add generated files to .gitignore

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
WIP - Reorganize and clean up pio man pages

- Move overview about PlatformIO
- Move descriptions for built in high-level aliases in PlatformIO

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz commented on service/docs/source/conf.py:
```diff
@@ -139,6 +139,7 @@ def get_description(rst_file_name):
     "geopm_imbalancer.3",
     "geopmlaunch.1",
     "geopm_pio_c.3",
+    "geopm_pio_sst.7",
```
Missing geopm_pio.7.rst

bgeltz commented on service/docs/source/geopm.7.rst:
```diff
@@ -387,6 +380,7 @@ SEE ALSO
 `geopm_agent_power_balancer(7) <geopm_agent_power_balancer.7.html>`_\ ,
 `geopm_agent_power_governor(7) <geopm_agent_power_governor.7.html>`_\ ,
 `geopm_report(7) <geopm_report.7.html>`_\ ,
+`geopm_pio_sst(7) <geopm_pio_sst.7.html>`_\ ,
```
Mission pio.7.rst

dannosliwcd commented on service/docs/source/geopm_pio_sst.7.rst:
```diff
@@ -0,0 +1,197 @@
+* ``SST::COREPRIORITY:ASSOCIATION``:
+  (Core scope) Returns the core's assigned CLOS.
+
+* ``SST::COREPRIORITY:n:WEIGHT``:
```
#2332 went in today to rename this to `SST::COREPRIORITY:n:PRIORITY` and scale it to the `[0,1]` range. The ordering is preserved in that change (so 0 is still assigned to the bucket that should get more budget).

avilcheslopez committed:
Add initial draft of chapter 7 MSRIOGroup man page

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

lhlawson committed:
Added GPU IO Group Docs

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

lhlawson commented:
Added GPU IO documentation from #2327.
Associated Tasks:
- [x] GPU High Level Alias rename based on #1671 
- [x] GPU LevelZero signal rename to remove GPUCHIP 
- [x] Matching of aggregation types across LevelZero, DCGM, and NVML where appropriate - https://github.com/geopm/geopm/issues/2360
- [x] Remove SYSMAN environment setting as discussed in #2205 once documentation is complete
- [x] Renaming of all signals in IO Group implementation to match documentation changes 

Non-doc changes that match the above are being addressed in #2378

lhlawson committed:
Added geopm_pio.7 to header exemption list

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

cmcantalupo commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,317 @@
+.. role:: raw-html-m2r(raw)
```
please remove

lhlawson committed:
Addressed arch review feedback on extraneous signals, requirements section, extra \ :

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

lhlawson commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,243 @@
+
+  *  ``Aggregation``: Average
+
+  *  ``Domain``: gpu
```
Capitalize consistently (i.e. Average and GPU or average or gpu)

lhlawson committed:
Fixed mixed capitalization issues and AFFINITIZATION incorrect domain

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

lhlawson committed:
Fixed simple type Avcerage for GPU_UTILIZATION

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

dannosliwcd commented:
Notes from a discussion about this:

How to resolve the redundancy issue, where controls will have duplicate documentation for their same-named signals:
-	have a section for all signals
-	have a section for all controls. Add a note that all controls can also be read.

Include info about when/where signals and controls appear to a user. E.g., GEOPM must be compiled with support for this feature, must be on hardware that supports it.

**TODO** in general: Remove m2r comments and annotations (e.g., “role” headers, or any strings that say “m2r”, backticks followed by backslash and space in our lists of definitions “`\ :” the backslash space is unnecessary)

lhlawson committed:
Moved GPU signal & control descriptions below names

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

dannosliwcd commented on service/docs/source/geopm_pio_sst.7.rst:
```diff
@@ -0,0 +1,197 @@
+
+The following example uses geopmread and geopmwrite command-line
+tools.  These steps can also be followed within an agent. Enabling
+steps are also in the ``SSTFrequencyGovernor``.
```
Probably shouldn't refer to that class in docs yet.

`SSTFrequencyGovernor` isn't targeted until 2.1. I'm also looking into whether I can take care of that in `FrequencyGovernor` instead.

dannosliwcd committed:
Tidy up machine-generated markup in CNLIOGroup(3)

dannosliwcd committed:
add geopm_pio(7) to the man page build output

dannosliwcd committed:
Use reST definition lists for alias definitions in geopm_pio(7)

dannosliwcd committed:
Add geopm_pio_cnl(7) documentation

dannosliwcd committed:
Add man section 7 PIO docs to the reference table of contents

lhlawson committed:
Updated LevelZero formatting

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

dannosliwcd commented on service/docs/source/geopm_pio.7.rst:
```diff
@@ -0,0 +1,200 @@
+interface to provide arbitrary signals and controls.
+
+A domain is a discrete component within a compute node where a signal
+or control is applicable.  For more information about the
+``geopm_domain_e`` enum and the hierarchical platform description see
+`geopm::PlatformTopo(3) <GEOPM_CXX_MAN_PlatformTopo.3.html>`_.  A
```
Some of the readers of man 7 geopm_pio may be end-users trying to understand which signals/controls they are interested in. Should this also mention `geopmread --domain`?

dannosliwcd replied:
Added a suggestion to see geopmread/write for CLI-based exploration of PlatformIO

dannosliwcd committed:
Add structured formatting data for the cnl man page

dannosliwcd committed:
Add cnl docs to MANIFEST.EXEMPT

dannosliwcd committed:
Remove all-caps usage from man rst headings. Sphinx takes care of man page capitalization for us.

dannosliwcd committed:
Add geopm_pio(7) to the see-also section in geopm(7)

dannosliwcd committed:
Update SST docs to use the new PRIORITY name and description instead of WEIGHT

dannosliwcd committed:
Enable autosectionlabel_prefix_document so auto-labeled sections have unique label names

avilcheslopez committed:
Move MSRIOGroup high-level aliases to geopm_pio.7 page

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

dannosliwcd reviewed (dismissed):
The `--geopm-record-filter` description in `geopmlaunch(1)` seems very long. See if we can shorten it a bit without hurting its usefulness, and offload some of the discussion to `geopm_pio_profile(7)`. That section-7 page should cover the EPOCH_COUNT, REGION_HASH, and other singals provided by the profile iogroup, so find a way to relate EPOCH_COUNT and REGION_HASH to the record filter.

Also specific questions related to that geopmlaunch option:

> The first parameter is the buffer size; the default if not provided is 100. The second parameter is the minimum stable period length in number of records. The third parameter is the stable period hysteresis factor. The fourth parameter is the unstable period hysteresis factor."

> 
> what is a buffer size?  is that in number of samples, bytes, seconds?

> 
> "stable period hysteresis factor"
?

> how is that normalized

dannosliwcd committed:
Use option lists instead of bulleted lists to document options in section 1 man pages

lhlawson committed:
Updated basic dcgm & nvml formatting

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

dannosliwcd committed:
Tidy up m2r-generated formatting in some pages

avilcheslopez committed:
Sort aliases in geopm_pio.7 alphabetically

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

dannosliwcd commented on service/docs/source/geopm_pio.7.rst:
```diff
@@ -0,0 +1,252 @@
+Descriptions Of High Level Aliases
+----------------------------------
+
+``BOARD_CPU_ENERGY`` (**TODO**: currently ``ENERGY_BOARD_CPU``)
+    Current total energy of CPU components, as measured by the server's board.
+    See :ref:`geopm_pio_cnl(7) <geopm_pio_cnl.7:Requirements>` for signal availability
+    requirements.
+
+``BOARD_CPU_POWER`` (**TODO**: currently ``POWER_BOARD_CPU``)
+    Current power consumption of CPU components, as measured by the server's
+    board. See :ref:`geopm_pio_cnl(7) <geopm_pio_cnl.7:Requirements>` for signal
+    availability requirements.
+
```
Remove

dannosliwcd commented on service/docs/source/geopm_pio.7.rst:
```diff
@@ -0,0 +1,252 @@
+    Total energy measured on the server's board. See :ref:`geopm_pio_cnl(7)
+    <geopm_pio_cnl.7:Requirements>` for signal availability requirements.
+
+``BOARD_MEMORY_ENERGY`` (**TODO**: currently ``ENERGY_MEMORY``)
+    Current total energy of memory components, as measured by the server's
+    board. See :ref:`geopm_pio_cnl(7) <geopm_pio_cnl.7:Requirements>` for signal
+    availability requirements.
+
+``BOARD_MEMORY_POWER`` (**TODO**: currently ``POWER_MEMORY``)
+    Current power consumption of memory components, as measured by the server's
+    board. See :ref:`geopm_pio_cnl(7) <geopm_pio_cnl.7:Requirements>` for signal
+    availability requirements.
+
```
Remove

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+
+Requirements
+~~~~~~~~~~~~
+To use the GEOPM LevelZero signals and controls GEOPM must be compiled against the oneAPI LevelZero libraries and must be run on a system with discrete GPUs supported by LevelZero.  In addition the user must export ZES_ENABLE_SYSMAN=1 as specified by the Intel oneAPI Level Zero Sysman documentation.  See the Sysman specification for more info on related environment variables and their usage.
```
I think it would be useful to explain how to compile GEOPM with L0 support via the ```--enable-levelzero``` configure time option.  If we have other docu about how configure works, a pointer here would be good.

bgeltz replied:
I think it would be useful to expand the bit about ``ZES_ENABLE_SYSMAN=1`` to show an example invocation in a ```.. code-block::``` section.  I would show the 2 different ways of doing this: via a single command that begins with setting the variable and separately via 2 commands with the first being the export and the second being ```geopmlaunch``` or similar.

lhlawson replied:
Agreed on both.  @cmcantalupo we removed the configure option since this was intended to be 'user' documentation.  Where are we putting 'developer' documentation?

bgeltz replied:
Add a tooltip box that spells out how to do this configure time enabling, with an explanation that a future release of GEOPM will have this enabled in an RPM so you do not have to do a source build.

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Format``: Double
+      *  ``Unit``: Hertz
+``LEVELZERO::GPU_POWER``
+    Average GPU power over 40ms (via geopmread) or 8 control loop iterations.  Derivative signal based on LEVELZERO::GPU_ENERGY
```
```suggestion
    Average GPU power over 40ms (via ``geopmread``) or 8 control loop iterations.  Derivative signal based on ``LEVELZERO::GPU_ENERGY``
```

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+-------
+
+``LEVELZERO::GPU_CORE_FREQUENCY_STATUS``
+    The current frequency of the GPU Compute Hardware
```
```suggestion
    The current frequency of the GPU Compute Hardware.
```

All of these descriptions should end with a period.  It looks like there are several spots in this IOGroup that have the trailing period absent.  This should also be reflected in the string that is presented with ``geopmread``.

We were doing this originally in i.e. the MSRIOGroup, but it seems recently added signals/controls also have the period missing there.  They should all be updated appropriately.

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Unit``: Hertz
+``LEVELZERO::GPU_CORE_FREQUENCY_MIN_AVAIL``
+    The minimum supported frequency of the GPU Compute Hardware
+
```
```suggestion


```
Is this supposed to be 2 \n's?  I've got opens on the style we want here so let's bring it up with the group.

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Domain``: GPU_CHIP
+      *  ``Format``: Double
+      *  ``Unit``: None
+``LEVELZERO::GPU_UNCORE_UTILIZATION``
```
```suggestion
``LEVELZERO::GPU_UNCORE_ACTIVITY``
```
This seems to be what we settled on during the arch meeting.

lhlawson replied:
I agree that GPU_UNCORE_ACTIVITY is the high level alias, but this is meant to represent the lower level Sysman 'engine activity utilization' counter https://spec.oneapi.io/level-zero/latest/sysman/api.html#_CPPv418zes_engine_stats_t.  

Is it worthwhile to keep this named as is and just alias it to GPU_UNCORE_ACTIVITY (similar to the DCGM::DRAM_ACTIVE being aliased to GPU_UNCORE_ACTIVITY)?

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+`geopm(7) <geopm.7.html>`_\ ,
+`geopm::IOGroup(3) <GEOPM_CXX_MAN_IOGroup.3.html>`_\ ,
+`geopmwrite(1) <geopmwrite.1.html>`_\ ,
+`geopmread(1) <geopmread.1.html>`_
```
```suggestion
`geopm(7) <geopm.7.html>`_,
`geopm_pio(7) <geopm_pio.7.html>`_,
`geopm::IOGroup(3) <GEOPM_CXX_MAN_IOGroup.3.html>`_,
`geopmwrite(1) <geopmwrite.1.html>`_,
`geopmread(1) <geopmread.1.html>`_
```

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+
+Requirements
+~~~~~~~~~~~~
+To use the GEOPM NVML signals and controls GEOPM must be compiled against the NVML libraries and must be run on a system with hardware supported by NVML.
```
Same comment here as with the L0 docu: we should spell out what it means to "compile GEOPM against NVML" for the configure time options and pointers to other relevant build time docu.

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Format``: Double
+      *  ``Unit``: Celsius
+``NVML::GPU_ENERGY_CONSUMPTION_TOTAL``
+    GPU energy consumptoin in joules since the driver was loaded
```
```suggestion
    GPU energy consumption in joules since the driver was loaded.
```

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Format``: Double
+      *  ``Unit``: None
+``NVML::GPU_POWER``:
+    GPU Power usage in watts
```
```suggestion
    GPU power usage in watts.
```

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+    Aliases to NVML::GPU_FREQUENCY_CONTROL
+
+
+See Also
```
Include a pointer to NVML documentation.  Many of the descriptions above allude to this.

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Aggregation``: Expect_same (note: this is incorrect.  File bug and fix)
+      *  ``Domain``: GPU
+      *  ``Format``: Double
+      *  ``Unit``: None
```
Pointer here to the NVML docu regarding this signal?

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Format``: Double
+      *  ``Unit``: None
+``NVML::GPU_PCIE_RX_THROUGHPUT``
+    GPU PCIE receive throughput in Bytes per Second over a 20 millisecond period
```
```suggestion
    GPU PCIE receive throughput in bytes per second over a 20 millisecond period.
```

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Format``: Double
+      *  ``Unit``: Bytes/Second
+``NVML::GPU_PCIE_TX_THROUGHPUT``
+    GPU PCIE transmit throughput in Bytes per Second over a 20 millisecond period
```
```suggestion
    GPU PCIE transmit throughput in bytes per second over a 20 millisecond period.
```

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Format``: Double
+      *  ``Unit``: Bytes/Second
+``NVML::GPU_CPU_ACTIVE_AFFINITIZATION``
+    GPU associated with the specified cpu as determined by querying active processes on the GPU.  If no GPUs map to the CPU -1 is returned.  If multiple GPU map to the CPU NAN is returned.
```
```suggestion
    GPU associated with the specified CPU as determined by querying active processes on the GPU.  If no GPUs map to the CPU -1 is returned.  If multiple GPU map to the CPU NAN is returned.
```

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Format``: Double
+      *  ``Unit``: Hertz
+
+Signal Aliases
```
This section needs to have its style aligned with other similar sections.  E.g. replace "aliases" with "maps", and put it all on a single line.  This also needs to be done for the L0 page.  We should hash out how we want this to look and then apply it to all the pages.

bgeltz commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,205 @@
+      *  ``Unit``: Watts
+
+Control Aliases
+~~~~~~~~~~~~~~~~
```
```suggestion
~~~~~~~~~~~~~~~
```

bgeltz commented on service/docs/source/geopm_pio_sst.7.rst:
```diff
@@ -0,0 +1,197 @@
+
+
+SST-CP: Core Power
+~~~~~~~~~~~~~~~~~~~
```
```suggestion
~~~~~~~~~~~~~~~~~~
```

bgeltz commented on service/docs/source/geopm_pio_dcgm.7.rst:
```diff
@@ -0,0 +1,94 @@
+
+Requirements
+~~~~~~~~~~~~
+To use the GEOPM DCGM signals and controls GEOPM must be compiled against the DCGM libraries and must be run on a system with hardware supported by DCGM.
```
Include the information about the relevant configure time flag.

bgeltz commented on service/docs/source/geopm_pio_dcgm.7.rst:
```diff
@@ -0,0 +1,94 @@
+      *  ``Unit``: None
+
+Signal Aliases
+~~~~~~~~~~~~~~~~
```
```suggestion
~~~~~~~~~~~~~~
```

bgeltz replied:
This section needs to be style aligned with the other pages.

bgeltz commented on service/docs/source/geopm_pio_cnl.7.rst:
```diff
@@ -0,0 +1,145 @@
+
+This IOGroup does not expose any controls.
+
+Signal Aliases
```
We need to decide the ordering of these sections.  Here, you've got the aliases at the end, just before the "See Also" section.

Lowren has the aliases follow the normal sections, i.e. Signals -> Signal Aliases -> Controls -> Control Aliases -> See Also.

dannosliwcd commented on service/docs/source/geopm_pio.7.rst:
```diff
@@ -0,0 +1,252 @@
+interface to provide arbitrary signals and controls.
+
+A domain is a discrete component within a compute node where a signal
+or control is applicable.  For more information about the
+``geopm_domain_e`` enum and the hierarchical platform description see
```
The linked page doesn't provide that information, but says to see `geopm_topo_c(3)` for information about `geopm_domain_e`

dannosliwcd commented on service/docs/source/geopm_topo_c.3.rst:
```diff
@@ -54,7 +54,7 @@ SYNOPSIS
+Domain Types
 ------------
 
 The ``geopm_domain_e`` enum defines a set of values that correspond to
```
This list is missing `GEOPM_DOMAIN_GPU_CHIP`.

lhlawson commented on service/docs/source/geopm_pio.7.rst:
```diff
@@ -0,0 +1,252 @@
+    Total power aggregated over the GPU package.
+
+``GPU_TEMPERATURE``
+    Average GPU temperature in degrees Celsius.
```
https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-temp-sensors-t
LevelZero exposes the maximum temperature across the device domain, or the minimum, but not the average.
https://spec.oneapi.io/level-zero/latest/sysman/api.html#zestemperaturegetstate

lhlawson replied:
Our LevelZero IO Group also does not currently expose a signal for this alias.

lhlawson replied:
```suggestion
    GPU temperature in degrees Celsius.
```

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Unit``: None
+
+Signal Aliases
+~~~~~~~~~~~~~~~~
```
```suggestion
~~~~~~~~~~~~~~
```

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Unit``: Hertz
+
+Control Aliases
+~~~~~~~~~~~~~~~~
```
```suggestion
~~~~~~~~~~~~~~~
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``CPU_FREQUENCY_MAX``
+    Aliases to MSR::TURBO_RATIO_LIMIT:MAX_RATIO_LIMIT_0
+
+``CYCLES_REFERENCE``
```
```suggestion
``CPU_CYCLES_REFERENCE``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``CYCLES_REFERENCE``
+    Aliases to MSR::FIXED_CTR2:CPU_CLK_UNHALTED_REF_TSC
+
+``CYCLES_THREAD``
```
```suggestion
``CPU_CYCLES_THREAD``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``CYCLES_THREAD``
+    Aliases to MSR::FIXED_CTR1:CPU_CLK_UNHALTED_THREAD
+
+``ENERGY_DRAM``
```
```suggestion
``DRAM_ENERGY``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``ENERGY_DRAM``
+    Aliases to MSR::DRAM_ENERGY_STATUS:ENERGY
+
+``ENERGY_PACKAGE``
```
```suggestion
``PACKAGE_ENERGY``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``ENERGY_PACKAGE``
+    Aliases to MSR::PKG_ENERGY_STATUS:ENERGY
+
+``INSTRUCTIONS_RETIRED``
```
```suggestion
``CPU_INSTRUCTIONS_RETIRED``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``INSTRUCTIONS_RETIRED``
+    Aliases to MSR::FIXED_CTR0:INST_RETIRED_ANY
+
+``POWER_PACKAGE_LIMIT``
```
```suggestion
``CPU_POWER_PACKAGE_LIMIT``
```
or
```suggestion
``CPU_PACKAGE_POWER_LIMIT``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``POWER_PACKAGE_LIMIT``
+    Aliases to MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT
+
+``POWER_PACKAGE_MAX``
```
```suggestion
``CPU_POWER_PACKAGE_MAX``
```
or
```suggestion
``CPU_PACKAGE_POWER_MAX``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``POWER_PACKAGE_MAX``
+    Aliases to MSR::PKG_POWER_INFO:MAX_POWER
+
+``POWER_PACKAGE_MIN``
```
```suggestion
``CPU_POWER_PACKAGE_MIN``
```
or
```suggestion
``CPU_PACKAGE_POWER_MIN``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``POWER_PACKAGE_MIN``
+    Aliases to MSR::PKG_POWER_INFO:MIN_POWER
+
+``POWER_PACKAGE_TDP``
```
```suggestion
``CPU_POWER_PACKAGE_TDP``
```
or
```suggestion
``CPU_PACKAGE_POWER_TDP``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``POWER_PACKAGE_TDP``
+    Aliases to MSR::PKG_POWER_INFO:THERMAL_SPEC_POWER
+
+``POWER_PACKAGE_TIME_WINDOW``
```
```suggestion
``CPU_POWER_PACKAGE_TIME_WINDOW``
```
or
```suggestion
``CPU_PACKAGE_POWER_TIME_WINDOW``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``POWER_PACKAGE_TIME_WINDOW``
+    Aliases to MSR::PKG_POWER_LIMIT:PL1_TIME_WINDOW
+
+``TIMESTAMP_COUNTER``
```
```suggestion
``CPU_TIMESTAMP_COUNTER``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``CPU_FREQUENCY_CONTROL``
+    Aliases to MSR::PERF_CTL:FREQ
+
+``POWER_PACKAGE_LIMIT``
```
```suggestion
``CPU_POWER_PACKAGE_LIMIT``
```
or
```suggestion
``CPU_PACKAGE_POWER_LIMIT``
```

bgeltz commented on service/docs/source/geopm_pio_msr.7.rst:
```diff
@@ -0,0 +1,145 @@
+``POWER_PACKAGE_LIMIT``
+    Aliases to MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT
+
+``POWER_PACKAGE_TIME_WINDOW``
```
```suggestion
``CPU_POWER_PACKAGE_TIME_WINDOW``
```
or
```suggestion
``CPU_PACKAGE_POWER_TIME_WINDOW``
```

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Format``: Double
+      *  ``Unit``: Seconds
+``LEVELZERO::GPU_CORE_FREQUENCY_CONTROL``
+    The last frequency request for the GPU Compute Hardware
```
Include info about setting this, and what the expectations are around reading, NaNs, etc.

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Format``: Double
+      *  ``Unit``: Watts
+``LEVELZERO::GPU_POWER_LIMIT_MAX_AVAIL``
+    The maximum supported power limit in Watts
```
Max supported by what?  Hardware?  Software?  Is this a theoretical limit?

bgeltz commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,252 @@
+      *  ``Format``: Double
+      *  ``Unit``: Hertz
+``LEVELZERO::GPU_POWER_LIMIT_DEFAULT``
+    Default power limit of the GPU in Watts
```
Is this the boot time value?  Or just the value if it's never been set?  Or both?

bgeltz committed:
Alphabetize pio pages in MANIFEST

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I85ae1f84beffd62a829a33bef2b32645e3765a10

bgeltz committed:
Alphabetize pio 7 pages in service Makefile

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I76676c93011cabb05578d9d9bedb9c8eaccf40bf

Commit:
Added chapter 7 TIMEIOGroup and CPUInfoIOGroup man pages.

Signed-off-by: Asma H Al-Rawi <asma.h.al-rawi@intel.com>

bgeltz committed:
Add chapter 7 page for ProfileIOGroup

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: Ic824f0cdb6ea121c26b20f37b2659841561a0e79

bgeltz committed:
Remove trailing whitespace

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I26cc67d220a3ba1df24b9b6242febfa359e151b5

bgeltz committed:
Refactor geopm_prof_c.3

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: Ia77795623cde7846a36fc5fe541f976ce723cbf6

dannosliwcd committed:
Tidy up m2r formatting of geopm_report.7 and add notes on aggregation

- m2r: unnecessary token separators, malformed tables, malformed code
  snippets, html-specific cross-references
- aggregation: separately discuss time- and domain-aggregation

dannosliwcd committed:
Fix some cross-references in geopmlaunch, fix some typos, WIP reduce verbosity of some options where details can go in other manpage sections

dannosliwcd committed:
Make the CNL docs have bulleted lists of properties with lowercase values

dannosliwcd committed:
Remove high level alias documentation for board-CPU and board-memory signals

dannosliwcd committed:
Start a GEOPM documentation linter. Run sphinx-build -M geopmlint

dannosliwcd committed:
Use :doc: cross-references instead of hyperlinks so that sphinx can warn us when those links are broken

dannosliwcd committed:
Resolve broken doc references

dannosliwcd committed:
Fix broken label references

dannosliwcd committed:
Add documentation for FFILibrary, referencing the opened C library and the CFFI documentation

dannosliwcd committed:
Use intersphinx to generate references to external sphinx documentation

dannosliwcd commented:
A lot of the section 3 man pages have malformed function argument names. It looks like it's the names that have underscores in them.

e.g., `_signal\ *name*` should probably be `*signal_name*`

dannosliwcd committed:
Remove the empty and unreferenced analysis.rst file

dannosliwcd committed:
Clean up remaining sphinx warnings

- Resolves #2090

dannosliwcd committed:
Fail the build if sphinx emimts warnings

- Related: #2090
- Should prevent that issue from reappearing

dannosliwcd committed:
Make sphinx act extra nitpicky so it complains about broken references

dannosliwcd committed:
Add a docs_geopmlint target, but do not make it part of make docs

- This is just a proposed build target for a geopm documentation linter.
  Nobody asked for it, so it isn't added to the build-blocking path at
  this time.

dannosliwcd committed:
Fix failing checks for license headers

bgeltz committed:
Address review feedback

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I9c5aa47f98b91ce1dcd1cd0bbb18544fd89b92bf

dannosliwcd committed:
Fix external links

- Remove or replace broken links
- Update links with new canonical URLs
- Convert hyperlinks for our own documentation to sphinx references

bgeltz committed:
Add libgeopmd dependency to docs

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I24cb95c5d21c29f84184b65cc653a03736ea9d35

bgeltz committed:
Add geopmlint.py to EXTRA_DIST

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I89340de5645bbd86a053c8bfd0f981d6bd13301b

bgeltz committed:
Refactor of cpuinfo page

- Aligned with current style from the profile
  page.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: Idf94b8fd594aa95b9e26e5723c5a241e1a92897d

bgeltz committed:
Style alignment

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: Ib5d6b0d71a2aaf7e7427e6c074bc041a87440596

dannosliwcd committed:
Capitalize GEOPM when it refers to the project or high-level project components

avilcheslopez committed:
Rename MSRIOGroup high-level aliases

Rename aliases to follow PlatformIO high-level alias naming convention.

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

avilcheslopez committed:
Updated formatting of geopm_pio_msr.7 page

Updated formatting to be consistent with the rest of the chapter 7
documentation.

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

bgeltz committed:
Update topo cache text to reflect implementation

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: Ie8c180a6f1a045b7216f3d8812a188c2ef5fb23f

bgeltz committed:
Refactor geopmread, write,  and topo_c

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I1335ae474fefe23bc5ebed0df094f9e53bf84aa9

bgeltz committed:
Add ServiceIOGroup chapter 7 page

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I20f5e4725571f7a57370f535458ce5fac1dfa4a1

dannosliwcd committed:
Add geopmpy interfaces to the rst documentation

- Also modifies some docstrings in geopmpy files to resolve sphinx warnings
- Adds sphinx.ext.todo since one docstring uses the Todo directive.
  Keeps the default behavior, which strips out the Todo items before
  generating the documentation build output.
- Adds pandas to the intersphinx config since some geopmpy interfaces
  refer to pandas object types

dannosliwcd committed:
Add brief descriptions to env vars in geopm.7 and xref to full descriptions in geopmlaunch.1

dannosliwcd committed:
First pass at geopm.7. Fill in most of the section stubs (except misc)

dannosliwcd committed:
Add geopm_pio_service.7.rst to manifest.exempt

avilcheslopez committed:
Add missing MSRIOGroup aliases to chapter 7 pages

Added: CPU_FREQUENCY_MAX_CONTROL, CPU_FREQUENCY_MIN_CONTROL

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

dannosliwcd committed:
Partially revert 513c51632 to remove the make docs dependency on geopmpy

- Remove geopmpy API documentation from the geopmpy man page
- Remove cross-references to geopmpy API documentation

Krebrov001 commented on service/docs/source/geopm_pio.7.rst:
```diff
@@ -0,0 +1,220 @@
+    Target minimum operating frequency of the CPU based on the control
+    register.
+
+``CPU_FREQUENCY_STATUS``
+    Average CPU frequency over the specified domain.
+
+``CPU_FREQUENCY_STATUS``
+    The current operating frequency of the CPU.
+
```
There are two of them.

Krebrov001 replied:
Keep the second one.

Krebrov001 committed:
Add the high level aliases to the `PlatformIO::m_signal_descriptions`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

dannosliwcd committed:
Mock out geopmdpy.gffi during documentation builds and remove geopm compiled lib dependencies from documentation builds

Krebrov001 committed:
Use the second description of `CPU_FREQUENCY_STATUS`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

dannosliwcd committed:
Modify PlatformIOTest::signal_control_description to consider high-level alias overrides

dannosliwcd committed:
Add autodocs for geopmpy

dannosliwcd committed:
Add a quick-start section to the HPC runtime guide

bgeltz committed:
Fix docs build

- Remove geopmpy.version from the geopmpy.7 page
  as it is a generated file.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I1c79a3ae72af7a78f6107cb55eb55312c85b5501

bgeltz committed:
Add missing docu to build files

- New man pages were missing from dist_man_MANS,
  conf.py, and geopm-service.spec.in.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I358a19396b00b7ee8c57e5e35f559e0436fc8819

bgeltz committed:
Add documentation style guide

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I075bc12d6dcf371f4370948098e0977f91115381

dannosliwcd committed:
Tidy up links and installation dependencies in the runtime guide

- Resolves #2376

dannosliwcd committed:
Add docu.rst to the exempt manifest

dannosliwcd committed:
Add geopm_pio_*.7 section rules to the doc linter

dannosliwcd committed:
Add an illustration of the main user-facing features of the GEOPM HPC runtime to the runtime docs

dannosliwcd committed:
Fix some words that popped up on a spell-checker, and wrap some verbatim terms in double-backticks

dannosliwcd committed:
Add a geopm-msr-json sphinx directive to document low-level MSR signals and controls

dannosliwcd committed:
Extract MSR json descriptions to separate json files, and generate their .cpp files at build time

dannosliwcd committed:
Add generated documentation for signals and controls in geopm_pio_msr(7)

- Documentation is generated from our msr_data_*.json files

dannosliwcd committed:
add our msr json files to the exempt manifest

dannosliwcd committed:
Fix the generated MSR signal/control documentation to MSR::MSRNAME:FIELDNAME

dannosliwcd committed:
Add build generator inputs to EXTRA_DIST

dannosliwcd committed:
Document the msr documentation generator and the geopm documentation linter in docu.rst

dannosliwcd committed:
Emit an error if MSR documentation tries to read malformed json

dannosliwcd committed:
Add descriptions for TRL MSRs

dannosliwcd committed:
Add documentation for the RAPL_POWER_UNIT MSR

lhlawson committed:
Addressed review feedback on NVML, DCGM, and LevelZero docs

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

Krebrov001 committed:
Refactor `geopm_hash.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_imbalancer.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_pio_c.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopmread.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
efactor `geopmwrite.1.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_policystore_c.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

lhlawson commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,251 @@
+Requirements
+^^^^^^^^^^^^
+
+To use the GEOPM LevelZero signals and controls GEOPM must be compiled against the oneAPI LevelZero libraries and must be run on a system with discrete GPUs supported by LevelZero.  To compile against the oneAPI LevelZero libraries geopm must be configured using the --enable-levelzero flag.  The optional --with-liblevelzero flag may be used to indicate the path of the required libraries.  In addi ...
```
```suggestion
To use the GEOPM LevelZero signals and controls GEOPM must be compiled against the oneAPI LevelZero libraries and must be run on a system with discrete GPUs supported by LevelZero.  To compile against the oneAPI LevelZero libraries geopm must be configured using the --enable-levelzero flag.  The optional --with-levelzero flag may be used to indicate the path of the required libraries.  In addition the user must export ZES_ENABLE_SYSMAN=1 as specified by the Intel oneAPI Level Zero Sysman documentation.  See the Sysman specification for more info on related environment variables and their usage.
```

lhlawson commented on service/docs/source/geopm_pio_nvml.7.rst:
```diff
@@ -0,0 +1,206 @@
+Requirements
+^^^^^^^^^^^^
+
+To use the GEOPM NVML signals and controls GEOPM must be compiled against the NVML libraries and must be run on a system with hardware supported by NVML.  To compile against the NVML libraries geopm must be configured using the --enable-nvml flag.  The optional --with-libnvml flag may be used to indicate the path of the required libraries.
```
```suggestion
To use the GEOPM NVML signals and controls GEOPM must be compiled against the NVML libraries and must be run on a system with hardware supported by NVML.  To compile against the NVML libraries geopm must be configured using the --enable-nvml flag.  The optional --with-nvml flag may be used to indicate the path of the required libraries.
```

lhlawson commented on service/docs/source/geopm_pio_dcgm.7.rst:
```diff
@@ -0,0 +1,95 @@
+Requirements
+^^^^^^^^^^^^
+
+To use the GEOPM DCGM signals and controls GEOPM must be compiled against the DCGM libraries and must be run on a system with hardware supported by DCGM.To compile against the oneAPI LevelZero libraries geopm must be configured using both the --enable-nvml flag and the --enable-dcgm flag.  The optional --with-libdcgm flag may be used to indicate the path of the required libraries.
```
```suggestion
To use the GEOPM DCGM signals and controls GEOPM must be compiled against the DCGM libraries and must be run on a system with hardware supported by DCGM.To compile against the oneAPI LevelZero libraries geopm must be configured using both the --enable-nvml flag and the --enable-dcgm flag.  The optional --with-dcgm flag may be used to indicate the path of the required libraries.
```

dannosliwcd committed:
Add descriptions for PKG_POWER_LIMIT MSRs

dannosliwcd committed:
Add descriptions for DRAM_POWER_LIMIT MSRs

dannosliwcd committed:
Add descriptions for DRAM_POWER_INFO MSRs

dannosliwcd committed:
Add descriptions for DRAM_PERF_STATUS MSRs

dannosliwcd committed:
Add descriptions for thermal MSRs

dannosliwcd committed:
Remove MSR::PLATFORM_INFO:PROGRAMMABLE_TCC_ACTIVATION_OFFSET from snb, where it does not exist

dannosliwcd committed:
Get rid of placeholders related to issue #2380

lhlawson commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,251 @@
+    *  **Unit**: joules
+
+``LEVELZERO::GPU_ENERGY_TIMESTAMP``
+    Timestamp for the GPU energy read in seconds.
```
```suggestion
    Timestamp for the GPU energy read in seconds.  Value is updated on LEVELZERO::GPU_ENERGY read.
```

Krebrov001 committed:
Refactor `GEOPM_CXX_MAN_PlatformIO.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

cmcantalupo committed:
Get rid of unused raw-html-m2r macro

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add descriptions for programmable counter MSRs

cmcantalupo committed:
Add a script to update all doxygen targets

- Remove Makefile scripting for doxygen
- Script copies output into sphinx build output directory
- Script can be run without any use of build system.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add missing package and core thermal interrupt MSR fields

- Resolves #2380
- Add the threshold fields for the core and package thermal interrupt
  MSRs.
- The added MSR fields can be written, but this change only adds
  signals. The purpose of this change is to ensure that the thermal
  status MSRs can be interpreted, and a user must know the value of
  the thermal interrupt cutoffs to fully interpret the thermal status
  MSRs.

dannosliwcd committed:
Rephrase some thermal status descriptions to be more direct about higher temperatures vs higher thermal readouts (which decrease as temperature approaches TCC)

dannosliwcd committed:
Automatically cross-reference MSR definitions when an MSR is mentioned in another MSR description

dannosliwcd committed:
Add a license header to doxygen_update.sh

dannosliwcd committed:
Remove some throwaway/test code I accidentally pushed

- I was exploring how to render MSR cross-references differently based
  on whether they were already present in the documentation.
- I went with an alternative route: require that we document all MSR
  fields that are mentioned in other MSR descriptions.

cmcantalupo committed:
Remove SKX signal and control list

- This is now documented in the IOGroup chapter 7 pages.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add the new thermal MSR signals to the legacy allowlist

- Related to #2380

dannosliwcd committed:
Remove SKX signal and control list from MANIFEST.EXEMPT

Krebrov001 committed:
Refactor `geopm_report.7`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_sched.3`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_version.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Refactor `geopm_time.3.rst`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

dannosliwcd committed:
Add doxygen_update.sh to EXTRA_DIST

cmcantalupo committed:
Add special section to reference.rst for PlatformIO documentation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add uncore MSR descriptions

cmcantalupo committed:
Clean up some of the path discovery in conf.py

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove _c from all C interface man page names

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

lhlawson committed:
Renamed DCGM & NVML signals, removed erroneous -lib from configure instructions

Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

cmcantalupo committed:
Use more globbing to make reference.rst easier to maintain

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get rid of erronous .h extensions in man page names

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove "index" from reference page

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd committed:
Add descriptions for PLATFORM_INFO MSRs

dannosliwcd committed:
Add descriptions for maximum package power limit time windows

dannosliwcd committed:
Describe ACNT, MCNT, and PCNT

dannosliwcd committed:
Add descriptions for FIXED_CTR_CTRL bits

dannosliwcd committed:
add a missing character in msr_data_skx

dannosliwcd committed:
Add descriptions for remaining global counter controls

dannosliwcd committed:
Add descriptions for MISC_ENABLE MSR bits

dannosliwcd committed:
Remove some accidentally duplicated words

dannosliwcd committed:
Add descriptions for MSRs related to bandwidth and cache monitoring by RMID

dannosliwcd committed:
Add alias descriptions for the QM counter aliases

dannosliwcd committed:
Make missing MSR descriptions a documentation-build-breaker

- The goal of this change is to prevent future additions of MSRs unless
  they are accompanied by documentation.

dannosliwcd committed:
Mention geopmread/write as command-line interfaces to PIO

dannosliwcd committed:
Removed a note about SSTFrequencyGovernor which does not target the same release as this version of documentation

dannosliwcd committed:
Link to geopm_topo C instead of C++ to explain geopm_domain_e

dannosliwcd committed:
Add GEOPM_DOMAIN_GPU_CHIP to the geopm_topo.3 documentation

dannosliwcd commented:
I combed through my stale comments on this PR. One that partially remains is the out-of-order words with underscores. (https://github.com/geopm/geopm/pull/2329#issuecomment-1144983682)

`grep '_\w\+\\ \*' docs/source/*rst` at least catches some remaining instances of that problem in `geopmbench.1`, `GEOPM_CXX_MAN_MSRIOGroup.3` and `GEOPM_CXX_MAN_SampleAggregator.3`

cmcantalupo committed:
Perform all remaining renames based on documentation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove high level signals from CNL IOGroup

- These signals are measured from a board level sensor
  but apply only to a subdomain on the board.
- These aliases do not follow our naming convention
  and will be confusing to the reader.
- Low level signals can be used instead.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

lhlawson commented on integration/test_skipped/test_levelzero_signals.py:
```diff
@@ -82,9 +82,9 @@ def test_energy(self):
-        frequency_gpu = geopm_test_launcher.geopmread("LEVELZERO::GPUCHIP_FREQUENCY gpu 0")
-        gpu_min_frequency_limit = geopm_test_launcher.geopmread("LEVELZERO::GPUCHIP_FREQUENCY_MIN_AVAIL gpu 0")
-        gpu_max_frequency_limit = geopm_test_launcher.geopmread("LEVELZERO::GPUCHIP_FREQUENCY_MAX_AVAIL gpu 0")
+        frequency_gpu = geopm_test_launcher.geopmread("LEVELZERO::GPU_CORE_FREQUENCY gpu 0")
```
Should be GPU_CORE_FREQUENCY_STATUS

lhlawson commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,251 @@
+^^^^^^^^^^^^^^
+
+``GPU_ENERGY``
+    Maps to ``LEVELZERO::ENERGY``.
```
```suggestion
    Maps to ``LEVELZERO::GPU_ENERGY``.
```

lhlawson commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,251 @@
+    *  **Format**: double
+    *  **Unit**: joules
+
+``LEVELZERO::ENERGY_TIMESTAMP``
```
```suggestion
``LEVELZERO::GPU_ENERGY_TIMESTAMP``
```

lhlawson commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,251 @@
+    *  **Format**: double
+    *  **Unit**: hertz
+
+``LEVELZERO::ENERGY``
```
```suggestion
``LEVELZERO::GPU_ENERGY``
```

lhlawson commented on service/docs/source/geopm_pio_levelzero.7.rst:
```diff
@@ -0,0 +1,251 @@
+    *  **Unit**: seconds
+
+``LEVELZERO::GPU_POWER``
+    average GPU power over 40ms (via geopmread) or 8 control loop iterations.  Derivative signal based on ``LEVELZERO::ENERGY``.
```
```suggestion
    average GPU power over 40ms (via geopmread) or 8 control loop iterations.  Derivative signal based on ``LEVELZERO::GPU_ENERGY``.
```

cmcantalupo committed:
Update power related signal and control names

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup rename for LevelZero energy metrics

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fix TODO for CPUINFO:MAX_FREQ

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Get rid of empty section of the MPIComm class documentation

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup missed renames in CNLIOGroup for memory signals

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
